app/controllers/admin/visit_request/check_in_controller.rb

```diff
@@ -0,0 +1,54 @@
+module Admin
+  class VisitRequest
+    class CheckInController < ::Admin::BaseController
+      helper_method :event, :upcoming_event, :user, :visit_request, :visit_count
+
+      def show
+        begin
+          ::VisitRequest::CheckIn.new(visit_request).call
+          render :show
+        rescue ActiveRecord::RecordNotFound
```

Comment:
looks to defensife. Maybe to not use exceptions for control flow? Just `:symbols` as status?

Reply:
```ruby
def show
  # Just return 2 values from this service. 1st as visit_request object and 2nd as symbol status
  # in such way your controller would not know so much about logic
  @visit_request, status = ::VisitRequest::CheckIn.new(visit_request).call
  render status
end
```